plugins/inputs/systemd_units/subcommand_show.go

```diff
@@ -0,0 +1,149 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package systemd_units
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+)
+
+type valueDef struct {
+	valueName string
+	valueMap  *map[string]int
+}
+
+// The following two maps configure the mapping of systemd properties to
+// tags or values. The properties are automatically requested from
+// `systemctl show`.
+// If a vlaue has no valueMap, `atoi` is called on the value to convert it to
+// an integer.
+var tagMap = map[string]string{
+	"Id":             "name",
+	"LoadState":      "load",
+	"ActiveState":    "active",
+	"SubState":       "sub",
+	"UnitFileState":  "uf_state",
+	"UnitFilePreset": "uf_preset",
+}
+
+var valueMap = map[string]valueDef{
+	"LoadState":         {valueName: "load_code", valueMap: &loadMap},
+	"ActiveState":       {valueName: "active_code", valueMap: &activeMap},
+	"SubState":          {valueName: "sub_code", valueMap: &subMap},
+	"StatusErrno":       {valueName: "status_errno", valueMap: nil},
+	"NRestarts":         {valueName: "restarts", valueMap: nil},
+	"MemoryCurrent":     {valueName: "mem_current", valueMap: nil},
+	"MemoryPeak":        {valueName: "mem_peak", valueMap: nil},
+	"MemorySwapCurrent": {valueName: "swap_current", valueMap: nil},
+	"MemorySwapPeak":    {valueName: "swap_peak", valueMap: nil},
+	"MemoryAvailable":   {valueName: "mem_avail", valueMap: nil},
+	"MainPID":           {valueName: "pid", valueMap: nil},
+}
+
+// Gather parses systemctl outputs and adds counters to the Accumulator
+func parseShow(_ *SystemdUnits, buffer *bytes.Buffer, acc telegraf.Accumulator) error {
```

Comment:
Neither this function nor the `parseListUnits` function uses the `SystemdUnits` parameter nor does any of those return an error, so how about the following
```suggestion
func parseShow(acc telegraf.Accumulator, buffer *bytes.Buffer) {
```

Reply:
The unused `SystemdUnits` structure and return value are there to ease extending the parser functions. I thought that later on it may be necessary to access configuration options or return an error when parsing.

If this currently unused noise is a non-idiomatic, I'll remove the parameter and return value. Just wanted to make sure you understand my thoughts behind the decision to put them in.

Please give me a short feedback if these should be removed or kept.

Reply:
I understand. However, we can easily add those back later if need be. We already have way-too-many "this is for a future that never happens" things in the code-base. ;-)